operator/duties/attester.go

```diff
@@ -75,74 +51,52 @@ func (h *AttesterHandler) HandleDuties(ctx context.Context) {
+			if !h.network.PastAlanForkAtEpoch(epoch) {
+				if h.firstRun {
+					h.processFirstRun(ctx, epoch, slot)
+				}
+				h.processExecution(epoch, slot)
+				if h.indicesChanged {
+					h.processIndicesChange(epoch, slot)
+				}
+				h.processFetching(ctx, epoch, slot)
+				h.processSlotTransition(epoch, slot)
 			}
 
 		case reorgEvent := <-h.reorg:
-			currentEpoch := h.network.Beacon.EstimatedEpochAtSlot(reorgEvent.Slot)
-			buildStr := fmt.Sprintf("e%v-s%v-#%v", currentEpoch, reorgEvent.Slot, reorgEvent.Slot%32+1)
-			h.logger.Info("🔀 reorg event received", zap.String("epoch_slot_pos", buildStr), zap.Any("event", reorgEvent))
-
-			// reset current epoch duties
-			if reorgEvent.Previous {
-				h.duties.ResetEpoch(currentEpoch)
-				h.fetchCurrentEpoch = true
-				if h.shouldFetchNexEpoch(reorgEvent.Slot) {
-					h.duties.ResetEpoch(currentEpoch + 1)
-					h.fetchNextEpoch = true
-				}
+			epoch := h.network.Beacon.EstimatedEpochAtSlot(reorgEvent.Slot)
+			tickerID := fields.FormatSlotTickerID(epoch, reorgEvent.Slot)
+			h.logger.Info("🔀 reorg event received", fields.SlotTickerID(tickerID), zap.Any("event", reorgEvent))
 
-				h.processFetching(ctx, currentEpoch, reorgEvent.Slot)
-			} else if reorgEvent.Current {
-				// reset & re-fetch next epoch duties if in appropriate slot range,
-				// otherwise they will be fetched by the appropriate slot tick.
-				if h.shouldFetchNexEpoch(reorgEvent.Slot) {
-					h.duties.ResetEpoch(currentEpoch + 1)
-					h.fetchNextEpoch = true
-				}
+			if !h.network.PastAlanForkAtEpoch(epoch) {
+				h.processReorg(ctx, epoch, reorgEvent)
 			}
 
 		case <-h.indicesChange:
 			slot := h.network.Beacon.EstimatedCurrentSlot()
-			currentEpoch := h.network.Beacon.EstimatedEpochAtSlot(slot)
-			buildStr := fmt.Sprintf("e%v-s%v-#%v", currentEpoch, slot, slot%32+1)
-			h.logger.Info("🔁 indices change received", zap.String("epoch_slot_pos", buildStr))
-
-			h.fetchCurrentEpoch = true
+			epoch := h.network.Beacon.EstimatedEpochAtSlot(slot)
+			tickerID := fields.FormatSlotTickerID(epoch, slot)
+			h.logger.Info("🔁 indices change received", fields.SlotTickerID(tickerID))
 
-			// reset next epoch duties if in appropriate slot range
-			if h.shouldFetchNexEpoch(slot) {
-				h.duties.ResetEpoch(currentEpoch + 1)
-				h.fetchNextEpoch = true
+			if !h.network.PastAlanForkAtEpoch(epoch) {
+				h.indicesChanged = true
 			}
 		}
 	}
 }
 
-func (h *AttesterHandler) HandleInitialDuties(ctx context.Context) {
-	ctx, cancel := context.WithTimeout(ctx, h.network.Beacon.SlotDurationSec()/2)
-	defer cancel()
-
-	slot := h.network.Beacon.EstimatedCurrentSlot()
-	epoch := h.network.Beacon.EstimatedEpochAtSlot(slot)
+func (h *AttesterHandler) processFirstRun(ctx context.Context, epoch phase0.Epoch, slot phase0.Slot) {
+	h.fetchCurrentEpoch = true
 	h.processFetching(ctx, epoch, slot)
+
+	if uint64(slot)%h.network.Beacon.SlotsPerEpoch() > h.network.Beacon.SlotsPerEpoch()/2-1 {
+		h.fetchNextEpoch = true
+	}
```

Comment:
Can use `if h.shouldFetchNexEpoch(...` here ? Although it uses `>=` (would be nice to clarify which is the correct one).